File: tests/fixtures/haskell/sample.hs
module Sample (greet, add, Name, Shape) where

import Data.List (sort, nub)
import qualified Data.Map as Map

data Shape = Circle Double | Rectangle Double Double

newtype Wrapper a = Wrapper a

type Name = String

class Describable a where
  describe :: a -> String

instance Describable Shape where
  describe (Circle r) = "Circle " ++ show r
  describe (Rectangle w h) = "Rect " ++ show w ++ "x" ++ show h

greet :: Name -> String
greet name = "Hello, " ++ name ++ "!"

add :: Int -> Int -> Int
add a b = a + b

main :: IO ()
main = do
  let msg = greet "World"
  let total = add 1 2
  putStrLn msg

File: tests/fixtures/csharp/sample.cs
using System;
using System.Collections.Generic;

namespace Sample
{
    public class Greeter
    {
        private readonly string _name;

        public Greeter(string name)
        {
            _name = name;
        }

        public string Greet()
        {
            return $"Hello, {_name}!";
        }
    }

    public interface ICalculator
    {
        int Add(int a, int b);
    }

    public static class Program
    {
        public static void Main()
        {
            var greeter = new Greeter("World");
            Console.WriteLine(greeter.Greet());
        }
    }
}
